trulens_eval/tests/unit/test.py

```diff
@@ -50,6 +59,59 @@ def module_installed(module: str) -> bool:
 
 
 class JSONTestCase(TestCase):
+    """TestCase class that adds JSON comparisons and golden expectation handling."""
+
+    def assertGoldenJSONEqual(
+        self,
+        actual,
+        golden_filename: str,
+        skips=None,
+        numeric_places: int = 7,
+    ):
+        """Assert equality between JSON-like `actual` and the content of
+        `golden_filename`.
+
+        If the environment variable `WRITE_GOLDEN` is set, the golden file will
+        be overwritten with the `actual` content. See `assertJSONEqual` for
+        details on the equality check.
+        """
+
+        write_golden: bool = bool(os.environ.get("WRITE_GOLDEN", ""))
+
+        caller_path = Path(caller_frame(offset=1).f_code.co_filename).parent
+        golden_path = (caller_path / "golden" / golden_filename).resolve()
+
+        if write_golden:
+            with golden_path.open("w") as f:
+                if golden_path.suffix == ".json":
+                    json.dump(actual, f)
```

Comment:
Can you prettify it before dumping so that it's easier to see changes later? Same with `yaml` if that's necessary (though I imagine it's not?)

Reply:
Did this.